### PR TITLE
fix QgsApplication::setPrefixPath

### PR DESCRIPTION
**Pull request by 3nids**
setPrefixPath was useless outside of the app since mPrefixPath was overwritten by env variables

### CODE REVIEW

File: src/core/qgsapplication.cpp [hunk 1]
```diff
@@ -207,32 +207,35 @@ void QgsApplication::init( QString profileFolder )
   }
   else
   {
-    char *prefixPath = getenv( "QGIS_PREFIX_PATH" );
-    if ( !prefixPath )
+    if ( ABISYM( mPrefixPath ).isNull() )
```
Comment (3nids) [inline src/core/qgsapplication.cpp]: this is actually the only adding line in this PR.
The following lines are only executed if `mPrefixPath` was not set before by `setPrefixPath` which happens in a 3rd party application

**Comment by 3nids** [comment]
@elpaso @NathanW2 @wonder-sk if you have anything to say, say it now or be quiet forever :)

**Comment by elpaso** [comment]
If the only addition is a check for `if ( ABISYM( mPrefixPath ).isNull() )` before reading it from the environment I think it is safe change.

**Comment by nyalldawson** [comment]
Please don't merge before Nathan's commented - this app logic is super tricky and has non obvious side effects

File: src/core/qgsapplication.cpp (hunk 1 above)
Comment (m-kuhn) [reply src/core/qgsapplication.cpp]: I think this should go one level deeper (inside the `if ( !prefixPath )`), so the env variable has the priority

**Comment by m-kuhn** [comment]
@NathanW2 what do you think about this change?

**Comment by NathanW2** [comment]
Having a look today sorry. Just got lost.

File: src/core/qgsapplication.cpp (hunk 1 above)
Comment (m-kuhn) [reply src/core/qgsapplication.cpp]: @3nids what do you think about this suggestion? would be nice to merge this at some point.

Comment (3nids) [reply src/core/qgsapplication.cpp]: @m-kuhn done

**Comment by 3nids** [comment]
@NathanW2 let me know if you still want to look

**Comment by NathanW2** [comment]
Merge it and I will look for any issues once it is in the nightly.

**Pull request merged**